app/src/main/java/com/pileproject/drive/comm/BluetoothCommunicator.java

```diff
@@ -18,6 +18,7 @@
 
 import android.bluetooth.BluetoothDevice;
 import android.bluetooth.BluetoothSocket;
+import android.support.annotation.NonNull;
 
 import com.pileproject.drivecommand.model.com.ICommunicator;
 
```

Comment:
I really hate this kind of boilerplates.
Why not consider using [Google Guava](https://github.com/google/guava)'s [precondition package](https://github.com/google/guava/wiki/PreconditionsExplained).

The cons of using this is that APKs using this library tend to be fat because the library has tons of packages.
I didn't check that the proguard can reduce the size of the APKs, but I believe it is worth considering the use
of the library in future.

Anyway I should conclude that Android API is a mess.